build(docs-infra): update Angular framework/Components/CLI to latest versions

gkalpak opened the pull request:
The PR updates Angular framework to v10.0.2, Angular Components to v10.0.1 and Angular CLI to v10.0.1. See individual commits for more details.

Fixes #37699.

mary-poppins commented:
You can preview 1ec42dc at https://pr37898-1ec42dc.ngbuilds.io/.

mary-poppins commented:
You can preview 38f1f24 at https://pr37898-38f1f24.ngbuilds.io/.

mary-poppins commented:
You can preview 4641ee5 at https://pr37898-4641ee5.ngbuilds.io/.

petebacondarwin commented on goldens/size-tracking/aio-payloads.json:
```diff
@@ -3,7 +3,7 @@
       "uncompressed": {
         "runtime-es2015": 2987,
-        "main-es2015": 453213,
+        "main-es2015": 450880,
```
woohoo! 2.5k decrease?

gkalpak replied:
Yes :tada: (Part of it is the framework and part of it is Components.)

pkozlowski-opensource reviewed (approved):
LGTM

Reviewed for: size-tracking

pkozlowski-opensource reviewed (approved):
Reviewed-for: size-tracking

jelbourn commented on aio/src/app/shared/custom-icon-registry.spec.ts:
```diff
@@ -6,21 +6,27 @@ describe('CustomIconRegistry', () => {
     const mockHttp: any = {};
     const mockSanitizer: any = {};
     const mockDocument: any = {};
+    const mockErrorHandler: any = {};
```
I wouldn't block submission on this, but I would do
```ts
const fakeErrorHandler = {
  handleError: e => { console.error(e); },
};
```

gkalpak replied:
Updated.
(Also updated the other names from `mock*` to `fake*` (since they are more fakes than mocks - although arguably they are neither :grin:).

jelbourn commented on aio/src/app/shared/custom-icon-registry.spec.ts:
```diff
@@ -6,21 +6,27 @@ describe('CustomIconRegistry', () => {
-    const registry = new CustomIconRegistry(mockHttp, mockSanitizer, mockDocument, svgIcons);
+
+    const registry =
+        new CustomIconRegistry(mockHttp, mockSanitizer, mockDocument, mockErrorHandler, svgIcons);
```
FYI there's actually an "official" `FakeMatIconRegistry` in `@angular/material/icon/testing` now

gkalpak replied:
:+1:
<sub>For future reference, we are testing our custom `MatIconRegistry` here, so using `FakeMatIconRegistry` is not applicable.</sub>

jelbourn reviewed (approved):
LGTM

Reviewed-for: size-tracking

gkalpak committed:
build(docs-infra): update @angular/core to 10.0.2

This commit updates the version of Angular framework used in angular.io
to version 10.0.2. It also features a commit message with a 100+ chars
long body.

gkalpak committed:
build(docs-infra): update @angular/material to 10.0.1

This commit updates the version of Angular Components used in angular.io
to version 10.0.1. It also updates the angular.io app to adapt to
breaking changes.

gkalpak committed:
refactor(docs-infra): avoid Material style duplication warning

This commit removes some duplicate imports of Material themes and
theming-related utilities. While this change does not have any impact on
the size of the generated `styles.css` file, it silences a build warning
pointing to [Avoiding duplicated theming styles][1].

[1]: https://github.com/angular/components/blob/db4b0cd1bff0bbb1741d22975bcfe1069aea3a53/guides/duplicate-theming-styles.md

gkalpak committed:
build(docs-infra): update @angular/cli to 10.0.1

This commit updates the version of Angular CLI used in angular.io to
version 10.0.1. It also reverts some changes (namely commits 38dfbc775f1
and eee2fd22e0a) which were made due to an older bug that is fixed in
the latest version. See #37688 for more details.

Fixes #37699

gkalpak committed:
fixup! build(docs-infra): update @angular/material to 10.0.1

mary-poppins commented:
You can preview 574620f at https://pr37898-574620f.ngbuilds.io/.
